dockerfiles/ci-unified/Dockerfile

```diff
@@ -103,21 +103,15 @@ RUN ln -s "/usr/local/rustup/toolchains/nightly-${RUST_NIGHTLY_VERSION}-x86_64-u
 # generic ci | install wasm toolchain for the default stable toolchain
 RUN rustup target add wasm32-unknown-unknown
 
-# Install rv32e-nightly-2023-04-05
-RUN wget https://github.com/paritytech/rustc-rv32e-toolchain/releases/download/rust-riscv32em-nightly-2023-04-05-r0-x86_64-unknown-linux-gnu/rust-riscv32em-nightly-2023-04-05-r0-x86_64-unknown-linux-gnu.tar.xz
-RUN tar xfz rust-riscv32em-nightly-2023-04-05-r0-x86_64-unknown-linux-gnu.tar.xz
-RUN mv rust-riscv32em-x86_64-unknown-linux-gnu/riscv32em-nightly-2023-04-05-r0-x86_64-unknown-linux-gnu/ /usr/local/rustup/toolchains
-RUN rm -f rust-riscv32em-nightly-2023-04-05-r0-x86_64-unknown-linux-gnu.tar.xz
-
 # generic ci | install x86_64 musl toolchain for the default stable toolchain
 RUN apt-get install -y --no-install-recommends g++
 RUN rustup target add x86_64-unknown-linux-musl
 
 # generic ci | install common cargo tools
-RUN cargo install cargo-web wasm-pack cargo-deny cargo-spellcheck cargo-hack \
+RUN cargo install cargo-web wasm-pack cargo-deny cargo-hack \
                   mdbook mdbook-mermaid mdbook-linkcheck mdbook-graphviz mdbook-admonish mdbook-last-changed
 
-RUN cargo install cargo-nextest --locked
+RUN cargo install cargo-spellcheck cargo-nextest --locked
 
```

Comment:
It is good practise to avoid the multipliction of layers in a docker image.
So instead of:
```
RUN cargo install foo
RUN cargo install bar
```

the following is preferable:
```
RUN cargo install foo bar
```

If you prefer a mutiline option to keep things clear, you may use:
```
RUN cargo install \
            foo bar \
            baz
```

Reply:
just changed what was breaking, I think we should move all these install to use --locked.
Otherwise this could break for future version as I discovered here.